webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala

```diff
@@ -554,6 +554,10 @@ class Boot extends Loggable {
         ApplicationLogger.error(s"Error when trying to save the EventLog for application start: ${err.fullMsg}")
       case Right(_)  => ApplicationLogger.info("Application Rudder started")
     }
+
+    // release guard for promise generation awaiting end of boot
+    RudderConfig.policyGenerationBootGuard.succeed(()).runNow
```

Comment:
I would never have thought to put it there, i was more thinking to put it at the end of bootcheck. It is without doubt better the way you did